tests/unit/test-waiter-test.ts

```diff
@@ -1,22 +1,46 @@
+// @ts-ignore
 import { module, test } from 'qunit';
 import { Promise } from 'rsvp';
 import { TestWaiter, _reset, getWaiters, getPendingWaiterState } from 'ember-test-waiters';
 import MockStableError, { overrideError, resetError } from './utils/mock-stable-error';
 
-module('test-waiter', function(hooks) {
+module('test-waiter', function(hooks: NestedHooks) {
   hooks.afterEach(function() {
     _reset();
     resetError();
   });
 
-  test('test waiter can be instantiated with a name', function(assert) {
+  test('test waiter can be instantiated with a name', function(assert: Assert) {
     let name = 'my-waiter';
     let waiter = new TestWaiter(name);
 
     assert.equal(waiter.name, name);
   });
 
-  test('test waiters automatically register when beginAsync is invoked', function(assert) {
+  test('test waiters return a token from beginAsync when no token provided', function(assert: Assert) {
+    let waiter = new TestWaiter('my-waiter');
+
+    let token = <number>waiter.beginAsync();
```

Comment:
Why cast this to a number? I think we should change this around to assert the value is non-null and non-undefined (we don't care if its a number or not).

We also need some tests (or assertions in this test) to ensure `waiter.beginAsync() !== waiter.beginAsync()`.

Reply:
Seems reasonable. Good suggestions.

Reply:
this is a good place for a type guard
```ts
let token = waiter.beginAsync();
if (typeof token !== 'number') {
   throw new Error('Non-numeric token: ' + token);
}
// token will be of type number here

```

Reply:
This has been cleaned up as part of the test PR, #9

Reply:
We've removed the need to cast now.